boards/nucleo-l1/include/periph_conf.h

```diff
@@ -45,37 +47,14 @@ extern "C" {
  * @brief Timer configuration
  * @{
  */
-#define TIMER_NUMOF         (2U)
-#define TIMER_0_EN          1
-#define TIMER_1_EN          1
-
-/* Timer 0 configuration */
-#define TIMER_0_DEV_0       TIM2
-#define TIMER_0_DEV_1       TIM3
-#define TIMER_0_CHANNELS    4
-#define TIMER_0_PRESCALER   (32U)
-#define TIMER_0_MAX_VALUE   (0xffff)
-#define TIMER_0_CLKEN()     (RCC->APB1ENR |= (RCC_APB1ENR_TIM2EN | RCC_APB1ENR_TIM3EN))
-#define TIMER_0_ISR_0       isr_tim2
-#define TIMER_0_ISR_1       isr_tim3
-#define TIMER_0_IRQ_CHAN_0  TIM2_IRQn
-#define TIMER_0_IRQ_CHAN_1  TIM3_IRQn
-#define TIMER_0_IRQ_PRIO    1
-#define TIMER_0_TRIG_SEL    TIM_SMCR_TS_0
-
-/* Timer 1 configuration */
-#define TIMER_1_DEV_0       TIM4
-#define TIMER_1_DEV_1       TIM5
-#define TIMER_1_CHANNELS    4
-#define TIMER_1_PRESCALER   (32U)
-#define TIMER_1_MAX_VALUE   (0xffff)
-#define TIMER_1_CLKEN()     (RCC->APB1ENR |= (RCC_APB1ENR_TIM4EN | RCC_APB1ENR_TIM5EN))
-#define TIMER_1_ISR_0       isr_tim4
-#define TIMER_1_ISR_1       isr_tim5
-#define TIMER_1_IRQ_CHAN_0  TIM4_IRQn
-#define TIMER_1_IRQ_CHAN_1  TIM5_IRQn
-#define TIMER_1_IRQ_PRIO    1
-#define TIMER_1_TRIG_SEL    TIM_SMCR_TS_1
+static const timer_conf_t timer_config[] = {
+    /* device, RCC bit, IRQ bit */
+    {TIM5, 3, TIM5_IRQn},
+};
+/* interrupt routines */
+#define TIMER_0_ISR         (isr_tim5)
+/* number of defined timers */
+#define TIMER_NUMOF         (sizeof(timer_config) / sizeof(timer_config[0]))
```

Comment:
The constant will be computed by the compiler at compile time, there won't
be a division at run time.
On Sep 15, 2015 8:18 PM, "A-Paul" <notifications@github.com> wrote:

> In boards/nucleo-l1/include/periph_conf.h
> <https://github.com/RIOT-OS/RIOT/pull/3856#discussion_r39546009>:
>
> > -#define TIMER_1_MAX_VALUE   (0xffff)
> > -#define TIMER_1_CLKEN()     (RCC->APB1ENR |= (RCC_APB1ENR_TIM4EN | RCC_APB1ENR_TIM5EN))
> > -#define TIMER_1_ISR_0       isr_tim4
> > -#define TIMER_1_ISR_1       isr_tim5
> > -#define TIMER_1_IRQ_CHAN_0  TIM4_IRQn
> > -#define TIMER_1_IRQ_CHAN_1  TIM5_IRQn
> > -#define TIMER_1_IRQ_PRIO    1
> > -#define TIMER_1_TRIG_SEL    TIM_SMCR_TS_1
> > +static const timer_conf_t timer_config[] = {
> > +    /* device, RCC bit, IRQ bit */
> > +    {TIM5, 3, TIM5_IRQn},
> > +};
> > +/* interrupt routines */
> > +#define TIMER_0_ISR         (isr_tim5)
> > +/* number of defined timers */
> > +#define TIMER_NUMOF         (sizeof(timer_config) / sizeof(timer_config[0]))
>
> A division for every time referencing a constant value?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RIOT-OS/RIOT/pull/3856/files#r39546009>.
>